### PR TITLE
[Doc] Add Architecture Doc for Ray Serve

### PR DESCRIPTION
**Pull request by simon-mo**
This PR adds architecture doc for Ray Serve. This has been a common user ask and confusion. Having this doc would be great for user trying to understand how Serve works and what happens when its applied to their use case.

### CODE REVIEW

**Comment by simon-mo** [comment]
Figure rendered:

Serve Architecture For Doc

**Review by edoakes — APPROVED** [review]
LGTM. Let's wait for @anabranch to sign off as well.

File: doc/source/serve/architecture.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+- Router: There is one router per node. Each router is a `Uvicorn <https://www.uvicorn.org/>`_ HTTP
+  server that accepts incoming requests, forwards them to the worker replicas, and
+  responds once they are completed.
+- Worker Replica: Worker replicas actually execute the code in response to a
+  request. For example, they may contain an instantiation of an ML model. Each
+  replica processes individual requests or batches of requests from the routers.
+
+
+Deep Dive
+---------
```
Comment (anabranch) [inline doc/source/serve/architecture.rst:29]: can we rename this section to "lifetime of a request" or something similar?

**Review by bllchmbrs — APPROVED** [review]
Minor change, then LGTM

**Pull request merged**